File: SolidSamples_Part2_OCP/app/src/main/java/com/example/solidsamples/bad/java/AreaCalculator.java
package com.example.solidsamples.bad.java;

/**
 * Created by pieter.snyman on 11/13/17.
 */

public class AreaCalculator {

    public double area(Object[] shapes) {
        double area = 0;
        for (Object shape : shapes) {
            if (shape instanceof Square) {
                Square square = (Square) shape;
                area += square.getWidth() * square.getHeight();
            } else {
                Circle circle = (Circle) shape;
                area += circle.getRadius() * circle.getRadius() * Math.PI;
            }
        }

        return area;
    }
}

File: SolidSamples_Part2_OCP/app/src/main/java/com/example/solidsamples/bad/kotlin/AreaCalculator.kt
package com.example.solidsamples.bad.kotlin

/**
 * Created by pieter.snyman on 11/13/17.
 */

class AreaCalculator {

    fun area(shapes: Array<Any>): Double {
        var area = 0.0

        shapes.forEach {
            when (it) {
                is Square -> {
                    area += it.width * it.height
                }
                is Circle -> {
                    area += it.radius * it.radius * Math.PI
                }
            }
        }

        return area
    }
}
